### ISSUE TITLE
Mod management

### ISSUE THREAD

**Issue by AWildErin** [AWildErin/sdklauncher#2]
Allow users to create source mods and set them up correctly. Can probably make it download the template repo or something

**Comment by Trico-Everfire** [comment]
Would this be extra buttons or a separate window that'd allow you to configure the source mod?

**Comment by AWildErin** [comment]
separate window that'd allow you to config the source mod really

**Comment by Trico-Everfire** [comment]
for Chaos source mods or P2CE source mods? or just, any source game? idk this is called SDKLauncher, doesn't specify for what exactly, I assume Chaos?

**Comment by AWildErin** [comment]
Chaos only, specifically for p2ce though I suppose you could theoretically do it for any chaos games

**Comment by Trico-Everfire** [comment]
I mean, that's where the difficulty lies, if it's P2CE specific, we can likely ship a ZIP it unpacks with the source mod, if it's Chaos we need to find a way to fetch any chaos game that has modding support and append that to the list of sourcemod options somehow.

**Comment by Trico-Everfire** [comment]
unless I am just over complicating things greatly.

**Comment by AWildErin** [comment]
Or just download the template repo? There's a direct download url you can do and you can ultimately just make use of that

**Comment by Trico-Everfire** [comment]
for P2CE? doesn't that require git to clone?
even if it' directly downloaded, it'd still be a zip that needs unpacking.

**Comment by AWildErin** [comment]
No, it doesn't require git to clone. And while yes it'd need to be unpacked, it'd still be the most up-to-date rather than shipping a zip with each build.

**Comment by Trico-Everfire** [comment]
That's true. Luckily, QT provides a neat little network package that'd allow us to fetch and download it. it's gonna be quite a process to make work, but I can try my best.

**Comment by Trico-Everfire** [comment]
I have an idea for the best of both worlds, can it be done where chaos applications like P2CE and Momentum have a file in their root called `sourcemod` (no extension), it's a plain text file with a link to the sourcemod repo, this way we can scan for this file and fetch the sourcemod, and if there are chaos games in the future they can follow suit or leave the file out, if it can't be found, it won't show up as an option.

**Comment by Trico-Everfire** [comment]
I am curious, what library would be used to zip/unzip the zip files brought on by github?
